cpp/src/arrow/python/pandas_convert.cc

```diff
@@ -1203,6 +1203,48 @@ inline Status ConvertFixedSizeBinary(const ChunkedArray& data, PyObject** out_va
   return Status::OK();
 }
 
+inline Status ConvertStruct(const ChunkedArray& data, PyObject** out_values) {
+  PyAcquireGIL lock;
+  std::vector<OwnedRef> fields_data;
+  for (int c = 0; c < data.num_chunks(); c++) {
+    auto arr = static_cast<StructArray*>(data.chunk(c).get());
+    fields_data.resize(arr->fields().size());
```

Comment:
Is this always the same size for each chunk? If so, it might be more performant to get the field size just once outside the loop.